### PR TITLE
Adding option to run bootstrap with local copy

### LINKED ISSUE #282: bootstrap.sh  locally
we dont not have an option to bootstrap locally 

the script should have a flag / check to run locally or pull from source

+ add bootstrap from a specific branch / version

### PR DESCRIPTION
**Pull request by eifrach**
Signed-off-by: eifrach <eifrach@redhat.com>

# Description

* adding option to run a local copy of `bootstrap.sh`  

Fixes #282 

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: pipelines/bootstrap.sh [hunk 1]
```diff
@@ -163,16 +163,34 @@ function clean_openshift_pipelines() {
 
 }
 
-export BASEDIR=$(dirname "$0")
+
+if [[ "${0}" == *"bootstrap.sh" ]]; then
```
Comment (flaper87) [inline pipelines/bootstrap.sh:167]: I think you want to use something like: 

`SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )`

`${0}` is not guaranteed to be `bootstrap.sh` if you run `bash bootstrap.sh` or in some other form

File: pipelines/bootstrap.sh [hunk 2]
```diff
@@ -163,16 +163,34 @@ function clean_openshift_pipelines() {
...
+    echo "Running locally"
+    export WORKDIR=$PWD
+
+    if [[ $PWD == *'pipelines' ]]; then
```
Comment (flaper87) [inline pipelines/bootstrap.sh:171]: The suggested edit above should give you the path to the bootstrap.sh, from there you can use relative paths to build the `PIPELINES_DIR`

**Review by flaper87 — CHANGES_REQUESTED** [review]
Close enough! This will be useful for local-development and branches.

File: pipelines/bootstrap.sh [hunk 3]
```diff
@@ -163,26 +163,35 @@ function clean_openshift_pipelines() {
...
 export EDGE_DEPLOYER_ROLEBINDING=ztp-cluster-admin
 
-if [[ ${#} -ge 2 ]]; then
-    if [[ ${2} == 'clean' ]]; then
-        clean_openshift_pipelines
-        echo "Done!"
-        exit 0
-    fi
+
+if [[ ${@} == *'clean'* ]]; then
```
Comment (eifrach) [inline pipelines/bootstrap.sh:191]: @flaper87 i had to add this change

when running locally the clean is on $1 
when running remote its on $2

**Comment by alknopfler** [comment]
Please @eifrach when we finish to fix the issue with the mirror, rebase this and pass the check before merge it.

**Comment by eifrach** [comment]
@alknopfler @flaper87 
tested  and rebased to `main`

**Comment by alknopfler** [comment]
@eifrach please could you verify if the make file which invoke the bootstrap script should be modified also?
I think if the script is expecting a new param, should be modified also in the makefile

**Comment by alknopfler** [comment]
LGTM

**Review by flaper87 — APPROVED** [review]
/lgtm

**Pull request merged**